tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -44,29 +41,42 @@ astDeclarationsForLanguage language filePath = do
   pwd             <- runIO getCurrentDirectory
   let invocationRelativePath = takeDirectory (pwd </> currentFilename) </> filePath
   input <- runIO (eitherDecodeFileStrict' invocationRelativePath)
-  either fail (fmap (concat @[]) . traverse (syntaxDatatype language)) input
-
+  allSymbols <- runIO (getAllSymbols language)
+  either fail (fmap (concat @[]) . traverse (syntaxDatatype language allSymbols)) input
+
+-- Build a list of all symbols
+getAllSymbols :: Ptr TS.Language -> IO [(String, Named)]
+getAllSymbols language = do
```

Comment:
This replaces the `TS. ts_language_symbol_for_name` API entirely as we were having issues with there being multiple symbols for a name. Instead, we build a (compile time) list of symbols up front so that we can lookup `TSSymbol`s later.